From 3b5a4029df9b31bde4a9059338e3f14a63cfd772 Mon Sep 17 00:00:00 2001
From: Jason Hung <jhung@globalscaletechnologies.com>
Date: Fri, 20 Apr 2018 08:41:48 +0800
Subject: [PATCH 6/8] dts: add new espressobin dts for hw rev.7

Since ESPRESSObin hw rev.7 changed the ethernet port definition,
so we need to change the ethernet port mapping.

the port mapping on the board : |USB2|WAN|LAN0|LAN1|USB3|
---
 arch/arm64/boot/dts/marvell/Makefile               |  1 +
 .../boot/dts/marvell/armada-3720-community-v7.dts  | 65 ++++++++++++++++++++++
 2 files changed, 66 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-3720-community-v7.dts

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 9543788..7112995 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -8,6 +8,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-community.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-community-v5.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-community-v7.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-37x0-customer.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-axis.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-vd.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-community-v7.dts b/arch/arm64/boot/dts/marvell/armada-3720-community-v7.dts
new file mode 100644
index 0000000..f79055c
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-3720-community-v7.dts
@@ -0,0 +1,65 @@
+/*
+ * Device Tree file for Marvell Armada 3720 community board
+ * Copyright (C) 2016 Marvell
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This file is distributed in the hope that it will be useful
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "armada-3720-community.dts"
+
+/ {
+	model = "Marvell Armada 3720 Community Board rev.7";
+	compatible = "marvell,espressobin-rev7", "marvell,armada-3720-community", "marvell,armada3720", "marvell,armada3710";
+
+	dsa@0 {
+	/* for ESPRESSObin HW rev.7, we swapped the wan and lan0 port */
+		switch@0 {
+			port@1 {
+				reg = <1>;
+				label = "lan1";
+				phy-handle = <&phy1>;
+			};
+			port@3 {
+				reg = <3>;
+				label = "wan";
+				phy-handle = <&phy3>;
+			};
+		};
+	};
+};
-- 
1.9.1

